cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m

```diff
@@ -496,51 +486,38 @@ - (void)restartShadowsocks:(bool)configChanged {
     return;
   }
   if (configChanged || ![activeHostNetworkAddress isEqualToString:self.hostNetworkAddress]) {
-    DDLogInfo(@"Configuration or host IP address changed with the network. Restarting ss-local.");
+    DDLogInfo(@"Configuration or host IP address changed with the network. Reconnecting tunnel.");
     self.hostNetworkAddress = activeHostNetworkAddress;
-    [self.shadowsocks stop:^(ErrorCode errorCode) {
-      DDLogInfo(@"Shadowsocks stopped.");
-      self.shadowsocks.config = [self getShadowsocksNetworkConfig];
-      __weak PacketTunnelProvider *weakSelf = self;
-      [self.shadowsocks
-          startWithConnectivityChecks:true
-                           completion:^(ErrorCode errorCode) {
-                             ErrorCode clientErrorCode =
-                                 errorCode == noError || errorCode == udpRelayNotEnabled
-                                     ? noError
-                                     : errorCode;
-                             [weakSelf execAppCallbackForAction:kActionStart
-                                                      errorCode:clientErrorCode];
-                             if (clientErrorCode != noError) {
-                               DDLogWarn(@"Tearing down VPN");
-                               [self cancelTunnelWithError:
-                                         [NSError errorWithDomain:NEVPNErrorDomain
-                                                             code:NEVPNErrorConnectionFailed
-                                                         userInfo:nil]];
-                               return;
-                             }
-                             BOOL isUdpSupported = errorCode == noError;
-                             [self setConectionUdpSupport:isUdpSupported];
-                             [weakSelf.connectionStore save:self.connection];
-                           }];
-    }];
-  }
-}
-
-- (NSDictionary *)getShadowsocksNetworkConfig {
-  if (self.connection == nil) {
-    DDLogError(@"Failed to retrieve configuration from nil connection");
-    return nil;
+    long errorCode = noError;
+    NSError *err;
+    ShadowsocksCheckConnectivity(self.hostNetworkAddress, [self.connection.port intValue],
+                                 self.connection.password, self.connection.method, &errorCode,
+                                 &err);
+    __block ErrorCode clientErrorCode =
+        (errorCode == noError || errorCode == udpRelayNotEnabled) ? noError : errorCode;
+    if (err != nil) {
+      DDLogError(@"Failed to check connectivity. Tearing down VPN");
+      [self execAppCallbackForAction:kActionStart errorCode:clientErrorCode];
```

Comment:
I think this can use `errorCode` instead of `clientErrorCode` without any difference.

Reply:
Done.